viewport-segments-api/index.js

```diff
@@ -0,0 +1,54 @@
+const list = document.querySelector("ul");
+const detailHeading = document.querySelector(".detail-view h1");
+const postureOutput = document.querySelector(".posture-output");
+
+// Update detail view when list item is clicked on
+
+list.addEventListener("click", (event) => {
+  detailHeading.textContent = event.target.textContent;
+});
+
+// Output device posture when the posture changes
+
+function reportPostureOutput() {
+  postureOutput.textContent = `Device posture: ${navigator.devicePosture.type}`;
+  console.log("event fired");
+}
+
+reportPostureOutput();
+navigator.devicePosture.addEventListener("change", reportPostureOutput);
+
+// Output viewport segment details to each segment
+
+const wrapperElem = document.querySelector(".wrapper");
+const listViewElem = document.querySelector(".list-view");
+const detailViewElem = document.querySelector(".detail-view");
+
+function addSegmentOutput(segments, index, elem) {
```

Comment:
So, you want this to be executed whenever the configuration of the viewport change (this explains the bug where the panel doesn't show consistently). You need to listen to resize event (when the device is rotated) and update the value when the posture is changed but the device isn't rotated.

Reply:
I meant `window.addEventListener('resize', reportSegments);` and as you correctly wrote `navigator.devicePosture.addEventListener("change", reportSegments);`

Listening for the orientation is not a good way because for example on the Asus Zenbook Fold undocking/docking the keyboard will not change the orientation but the viewport will be resized.

Reply:
OK, so the `window.screen.orientation.addEventListener("change", reportSegments);` line is not needed?

What about when you rotate a device such as a smartphone, to change its orientation? does the `resize` handler pick that up? I wasn't sure if it would, as the screen area stays the same...but I guess, if the height and width are changing?